File: Java/src/main/java/hayaa/rpc/server/ProviderContainer.java
package hayaa.rpc.server;

import java.lang.reflect.Method;
import java.util.Hashtable;

/**
 * @author hsieh
 */
public class ProviderContainer {
    private String interfaceName;
    private Object serviceInstance;
    private Hashtable<String, Method> methods;

    public ProviderContainer(String interfaceName,Object serviceInstance, Hashtable<String, Method> methods) {
        this.interfaceName = interfaceName;
        this.serviceInstance=serviceInstance;
        this.methods = methods;
    }

    public Object getServiceInstance() {
        return serviceInstance;
    }

    public void setServiceInstance(Object serviceInstance) {
        this.serviceInstance = serviceInstance;
    }

    public String getInterfaceName() {
        return interfaceName;
    }

    public void setInterfaceName(String interfaceName) {
        this.interfaceName = interfaceName;
    }

    public Hashtable<String, Method> getMethods() {
        return methods;
    }

    public void setMethods(Hashtable<String, Method> methods) {
        this.methods = methods;
    }
}

File: C#/RPC/Hayaa.RPC/Server/ProviderContainer.cs
using System;
using System.Collections.Generic;
using System.Reflection;
using System.Text;

namespace Hayaa.RPC.Service.Server
{
  public  class ProviderContainer
    {
        public ProviderContainer(String interfaceName, Object serviceInstance, Dictionary<String, MethodInfo> methods)
        {
            this.InterfaceName = interfaceName;
            this.ServiceInstance = serviceInstance;
            this.Methods = methods;
        }
        public String InterfaceName { set; get; }
        public Object ServiceInstance { set; get; }
        public Dictionary<String, MethodInfo> Methods { set; get; }
    }
}
